indico/modules/users/controllers.py

```diff
@@ -435,6 +446,29 @@ def _process(self):
         return jsonify(self.user.settings.get('use_markdown_for_minutes'))
 
 
+class RHUserPreferencesMastodonServer(RHUserBase):
+    @use_kwargs({
+        'server_url': fields.String(required=True),
+    })
+    def _process_POST(self, server_url):
+        if not social_settings.get('enabled'):
+            raise Forbidden('The social share widget is not enabled.')
+
+        url = urlsplit(server_url)
+        if url.scheme not in ('http', 'https'):
+            raise ExpectedError(_('Invalid URL'))
+        url = urlunsplit(url)
```

Comment:
You're not removing anything - did you forget to add that? If not, you can just use `server_url` below, but I think you want to keep the path stripping logic...